pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -304,6 +304,28 @@ type MonitoringSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +optional
 	ExternalMgrPrometheusPort uint16 `json:"externalMgrPrometheusPort,omitempty"`
+	// AlertRuleOverrides points to a customized Ceph prometheus alerts
+	// +optional
+	// +nullable
+	AlertRuleOverrides map[string]*CephAlert `json:"alertRuleOverrides,omitempty"`
+}
+
+//CephAlert basic customized alert
+type CephAlert struct {
+	// +optional
+	Disabled bool `json:"disabled,omitempty"`
+	// +optional
+	For string `json:"for,omitempty"`
+	// +optional
+	SeverityLevel string `json:"severityLevel,omitempty"`
```

Comment:
If we have a specific list of severities, let's add the enum to the schema so we can enforce it without any other code, something like this:
```suggestion
	// +kubebuilder:validation:Enum=warning,error,critical
	SeverityLevel string `json:"severityLevel,omitempty"`
```

Reply:
Nice!

Reply:
Are the `severity` and `severityLevel` actually the same thing, just named differently for various rules? If we can't combine them, maybe we just need the enum on both.

Reply:
Regarding the suggestion about enum on both of the fields, severity and severityLevel are not exactly the same. severity is a label, so we can use a custom severity label in Prometheus. It might help if you would like to direct different rook alerts to different slack channels.

severityLevel must be enum because of ceph UI requirement (as can be seen here)

For example,
You can give some of the alerts the severity: rook-warning, so all the alerts with this label can be sent to a different channel that is separated from alerts with the severity: warning.

I believe that in severity field case, we should let the user decide which severity label to use.

Reply:
Ah ok, the `severity` really can be anything. How about a description on the settings that indicates the difference?

Reply:
Sure, I added notes in the docs that explain the difference.
Please let me know what so you think.